rpc/lib/server/http_server.go

```diff
@@ -31,15 +32,41 @@ const (
 	// same as the net/http default
 	maxHeaderBytes = 1 << 20
 
-	// Timeouts for reading/writing to the http connection.
-	// Public so handlers can read them -
-	// /broadcast_tx_commit has it's own timeout, which should
-	// be less than the WriteTimeout here.
+	// Timeout for reading to the http connection.
+	// Public so handlers can read it.
 	// TODO: use a config instead.
-	ReadTimeout  = 3 * time.Second
-	WriteTimeout = 20 * time.Second
+	ReadTimeout = 3 * time.Second
 )
 
+var (
+	// Timeout for writing to the http connection.
+	// Use the WriteTimeout() function to read this.
+	writeTimeout = 20 * time.Second
```

Comment:
Instead of having a global var here that gets adjusted, there could be a default (const) and we could pass in a `time.Duration` (TimeoutBroadcastTxCommit) into `StartHTTPServer`. If the passed time.Duration is bigger than the default the param passed into `http.Server` gets adjusted. That should be cleaner from what we do here. The problem with this approach is that we read the `WriteTimeout` in `rpc/core/mempool.go` and `rpc/core/events.go`. Also the comment suggests that handlers might want to read the value, too.